Test failures seem related:

```
2020-05-25T11:48:37.6183285Z [ERROR]   ArcEndpointTest.testBeans:58 expected: <io.quarkus.runtime.StartupEvent> but was: <io.vertx.ext.web.Router>
2020-05-25T11:48:37.6183630Z [INFO] 
```

--- a/docs/src/main/asciidoc/cdi-reference.adoc
+++ b/docs/src/main/asciidoc/cdi-reference.adoc
@@ -898,12 +898,12 @@ Here is a summary of which extensions can access which metadata:
 
 In the development mode, two special endpoints are registered automatically to provide some basic debug info in the JSON format:
 
-* HTTP GET `/arc/beans` - returns the list of all beans
+* HTTP GET `/quarkus/arc/beans` - returns the list of all beans
 ** You can use query params to filter the output:
-*** `scope` - include beans with scope that ends with the given value, i.e. `http://localhost:8080/arc/beans?scope=ApplicationScoped`
-*** `beanClass` - include beans with bean class that starts with the given value, i.e. `http://localhost:8080/arc/beans?beanClass=org.acme.Foo`
-*** `kind` - include beans of the specified kind (`CLASS`, `PRODUCER_FIELD`, `PRODUCER_METHOD`, `INTERCEPTOR` or `SYNTHETIC`), i.e. `http://localhost:8080/arc/beans?kind=PRODUCER_METHOD`
-* HTTP GET `/arc/observers` - returns the list of all observer methods
+*** `scope` - include beans with scope that ends with the given value, i.e. `http://localhost:8080/quarkus/arc/beans?scope=ApplicationScoped`
+*** `beanClass` - include beans with bean class that starts with the given value, i.e. `http://localhost:8080/quarkus/arc/beans?beanClass=org.acme.Foo`
+*** `kind` - include beans of the specified kind (`CLASS`, `PRODUCER_FIELD`, `PRODUCER_METHOD`, `INTERCEPTOR` or `SYNTHETIC`), i.e. `http://localhost:8080/quarkus/arc/beans?kind=PRODUCER_METHOD`
+* HTTP GET `/quarkus/arc/observers` - returns the list of all observer methods
 
 NOTE: These endpoints are only available in the development mode, i.e. when you run your application via `mvn quarkus:dev` (or `./gradlew quarkusDev`).
 

--- a/extensions/vertx-http/deployment/src/main/java/io/quarkus/vertx/http/deployment/devmode/ArcEndpointProcessor.java
+++ b/extensions/vertx-http/deployment/src/main/java/io/quarkus/vertx/http/deployment/devmode/ArcEndpointProcessor.java
@@ -6,15 +6,22 @@ import io.quarkus.deployment.annotations.BuildStep;
 import io.quarkus.deployment.annotations.ExecutionTime;
 import io.quarkus.deployment.annotations.Record;
 import io.quarkus.vertx.http.deployment.RouteBuildItem;
+import io.quarkus.vertx.http.runtime.HttpBuildTimeConfig;
 import io.quarkus.vertx.http.runtime.devmode.ArcEndpointRecorder;
 
 public class ArcEndpointProcessor {
 
     @Record(ExecutionTime.RUNTIME_INIT)
     @BuildStep(onlyIf = IsDevelopment.class)
-    void registerRoutes(ArcEndpointRecorder recorder, BuildProducer<RouteBuildItem> routes) {
-        routes.produce(new RouteBuildItem("/arc/beans", recorder.createBeansHandler()));
-        routes.produce(new RouteBuildItem("/arc/observers", recorder.createObserversHandler()));
+    void registerRoutes(HttpBuildTimeConfig config, ArcEndpointRecorder recorder, BuildProducer<RouteBuildItem> routes,
+            BuildProducer<NotFoundPageDisplayableEndpointBuildItem> displayableEndpoints) {
+        String basePath = config.consolePath + "/arc";
+        String beansPath = basePath + "/beans";
+        String observersPath = basePath + "/observers";
+        routes.produce(new RouteBuildItem(beansPath, recorder.createBeansHandler()));
+        routes.produce(new RouteBuildItem(observersPath, recorder.createObserversHandler()));
+        displayableEndpoints.produce(new NotFoundPageDisplayableEndpointBuildItem(beansPath));
+        displayableEndpoints.produce(new NotFoundPageDisplayableEndpointBuildItem(observersPath));
     }
 
 }

--- a/extensions/vertx-http/deployment/src/test/java/io/quarkus/vertx/http/devmode/ArcEndpointTest.java
+++ b/extensions/vertx-http/deployment/src/test/java/io/quarkus/vertx/http/devmode/ArcEndpointTest.java
@@ -5,6 +5,7 @@ import static org.junit.jupiter.api.Assertions.assertNotNull;
 
 import javax.enterprise.context.ApplicationScoped;
 import javax.enterprise.event.Observes;
+import javax.inject.Inject;
 import javax.inject.Named;
 
 import org.jboss.shrinkwrap.api.ShrinkWrap;
@@ -14,9 +15,11 @@ import org.junit.jupiter.api.extension.RegisterExtension;
 
 import io.quarkus.runtime.StartupEvent;
 import io.quarkus.test.QuarkusDevModeTest;
+import io.quarkus.vertx.http.runtime.HttpBuildTimeConfig;
 import io.restassured.RestAssured;
 import io.vertx.core.json.JsonArray;
 import io.vertx.core.json.JsonObject;
+import io.vertx.ext.web.Router;
 
 public class ArcEndpointTest {
 
@@ -27,8 +30,9 @@ public class ArcEndpointTest {
 
     @Test
     public void testBeans() {
-        JsonArray beans = new JsonArray(RestAssured.get("/arc/beans").asString());
-        JsonArray observers = new JsonArray(RestAssured.get("/arc/observers").asString());
+        String debugPath = RestAssured.get("/console-path").asString();
+        JsonArray beans = new JsonArray(RestAssured.get(debugPath + "/arc/beans").asString());
+        JsonArray observers = new JsonArray(RestAssured.get(debugPath + "/arc/observers").asString());
         JsonObject fooBean = null;
         JsonObject fooObserver = null;
         for (int i = 0; i < beans.size(); i++) {
@@ -46,20 +50,27 @@ public class ArcEndpointTest {
 
         for (int i = 0; i < observers.size(); i++) {
             JsonObject observer = observers.getJsonObject(i);
-            if (beanId.equals(observer.getString("declaringBean"))) {
+            if (beanId.equals(observer.getString("declaringBean"))
+                    && StartupEvent.class.getName().equals(observer.getString("observedType"))) {
                 fooObserver = observer;
+                assertEquals(2500, fooObserver.getInteger("priority"));
             }
         }
         assertNotNull(fooObserver);
-        assertEquals(StartupEvent.class.getName(), fooObserver.getString("observedType"));
-        assertEquals(2500, fooObserver.getInteger("priority"));
     }
 
     @Named
     @ApplicationScoped
     public static class Foo {
 
+        @Inject
+        HttpBuildTimeConfig httpConfig;
+
         void onStart(@Observes StartupEvent event) {
+        }
+
+        void addConfigRoute(@Observes Router router) {
+            router.route("/console-path").handler(rc -> rc.response().end(httpConfig.consolePath));
         }
 
     }

--- a/extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/HttpBuildTimeConfig.java
+++ b/extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/HttpBuildTimeConfig.java
@@ -21,4 +21,10 @@ public class HttpBuildTimeConfig {
      */
     @ConfigItem
     public boolean virtual;
+
+    /**
+     * The HTTP console path. Various debug/development endpoints are deployed under this path.
+     */
+    @ConfigItem(defaultValue = "/quarkus")
+    public String consolePath;
 }

